### PR TITLE
Added Group `tar`

### PR DESCRIPTION
**Pull request by RichardLitt**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1724,10 +1724,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
-## add
+This command can't be called directly.
+
+## add [POST /tar/add{?arg}]
+Import a tar file into IPFS
+
+'ipfs tar add' will parse a tar file and create a merkledag structure to represent it.
+
++ Parameters
+    + arg (string, required) - tar file to add
```
Comment (dignifiedquire) [inline apiary.apib]: Is this the file itself or the path to the tar file?

File: apiary.apib [hunk 2]
```diff
@@ -1724,10 +1724,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+    + Body
+
+        ```
+        {
+          "Name": "test",
+          "Hash": "QmUkwXfkw4mHH9PB53b5fR5C1YsjhLxz3LbjBehB9Q6LAG"
+        }
+        ```
+
++ Response 200 (text/plain)
```
Comment (dignifiedquire) [inline apiary.apib]: Why is there a second Response?

File: apiary.apib [hunk 3]
```diff
@@ -1724,10 +1724,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+
++ Request With Empty Argument
+
+    The response is the same if the argument is invalid. For example:
+
+        curl -i "http://localhost:5001/api/v0/tar/cat?arg=kitten"
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/tar/cat?arg="
```
Comment (dignifiedquire) [inline apiary.apib:3211]: Why are you not using backticks for this, like in the other places?

File: apiary.apib [hunk 4]
```diff
@@ -1724,10 +1724,261 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+        Date: Thu, 11 Feb 2016 22:14:37 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        input0000644000076500000240000000200012657203016012731 0ustar00richardstaff00000000000000
```
Comment (dignifiedquire) [inline apiary.apib:3322]: Is this the binary version of the tar file, or is this the untared version, like what format is is this

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib]: The file itself. How can I make that clearer?

File: apiary.apib (hunk 3 above)
Comment (RichardLitt) [inline apiary.apib:3211]: Backticks should only be used for the Body curl.

File: apiary.apib (hunk 4 above)
Comment (RichardLitt) [inline apiary.apib:3322]: I have no idea. It's what I got back in the CLI. cc @whyrusleeping

File: apiary.apib (hunk 1 above)
Comment (dignifiedquire) [inline apiary.apib]: Not sure :/ cc @diasdavid

File: apiary.apib (hunk 4 above)
Comment (RichardLitt) [inline apiary.apib:3322]: It's a .tar format. Known bug: https://github.com/ipfs/go-ipfs/issues/1824

**Comment by RichardLitt** [comment]
@diasdavid or @dignifiedquire: review?

**Comment by dignifiedquire** [comment]
LGTM

File: apiary.apib [hunk 5]
```diff
@@ -3070,9 +3072,255 @@ identifiers and resolves them to the referenced item.
...
+        ```
+        curl -i "http://localhost:5001/api/v0/tar/cat?arg=QmW2WQi7j6c7UgJTarActp7tDNikE4B2qXtFCfLPdsgaTQ"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
```
Comment (daviddias) [inline apiary.apib:3306]: Nice, no `ouptut` anymore,  I've noticed that in go-ipfs too :)

File: apiary.apib (hunk 4 above)
Comment (daviddias) [inline apiary.apib:3322]: maybe add that issue as as comment on the spec?

**Comment by daviddias** [comment]
LGTM overall :)

File: apiary.apib (hunk 5 above)
Comment (RichardLitt) [inline apiary.apib:3306]: Ahahaha I never noticed that.

**Pull request merged**